waspc/src/Wasp/Generator/ServerGenerator/JobGenerator.hs

```diff
@@ -67,16 +61,46 @@ genJobs spec = return $ genJob <$> getJobs spec
                   [ "jobName" .= jobName,
                     "jobPerformFnName" .= jobPerformFnName,
                     "jobPerformFnImportStatement" .= jobPerformFnImportStatement,
-                    "jobFactoryName" .= show (jobFactoryForJob job)
+                    "jobFactoryName" .= show (jobFactoryForJob $ executor job),
+                    "jobPerformOptions" .= jobPerformOptionsJsonString job
                   ]
             )
 
+-- | TODO: Make this not hardcoded!
+relPosixPathFromJobFileToExtSrcDir :: Path Posix (Rel (Dir ServerSrcDir)) (Dir GeneratedExternalCodeDir)
+relPosixPathFromJobFileToExtSrcDir = [reldirP|../ext-src|]
+
+jobFactoryForJob :: JobExecutor -> JobFactory
+jobFactoryForJob Passthrough = PassthroughJobFactory
+jobFactoryForJob PgBoss = PgBossJobFactory
+
 genJobFactories :: Generator [FileDraft]
 genJobFactories = return $ genJobFactory <$> jobFactories
   where
     genJobFactory :: JobFactory -> FileDraft
     genJobFactory jobFactory =
       let jobFactoryFp = jobFactoryFilePath jobFactory
           sourceTemplateFp = C.asTmplFile jobFactoryFp
-          destinationServerFp = C.asServerFile jobFactoryFp
-       in C.mkTmplFdWithDstAndData sourceTemplateFp destinationServerFp Nothing
+       in C.mkTmplFd sourceTemplateFp
+
+jobFactories :: [JobFactory]
+jobFactories = enumFrom minBound :: [JobFactory]
+
+jobFactoryFilePath :: JobFactory -> Path' (Rel d) File'
+jobFactoryFilePath PassthroughJobFactory = [relfile|src/jobs/PassthroughJobFactory.js|]
+jobFactoryFilePath PgBossJobFactory = [relfile|src/jobs/PgBossJobFactory.js|]
+
+isPgBossUsed :: AppSpec -> Bool
+isPgBossUsed spec = any (\(_, job) -> executor job == PgBoss) (getJobs spec)
+
+pgBossVersionBounds :: String
+pgBossVersionBounds = "^7.2.1"
+
+pgBossDependency :: AS.Dependency.Dependency
+pgBossDependency = AS.Dependency.make ("pg-boss", pgBossVersionBounds)
+
+maybePgBossDependency :: Maybe AppSpec -> Maybe AS.Dependency.Dependency
```

Comment:
I think it would be more standard to have signature `AppSpec -> Maybe AS.Dependency.Dependency`, so without that initial `Maybe`, and if somebody wants to use  this on `Maybe AppSpec`, they can by doing `>>=`.

Reply:
For some reason I feel a bit fishy about this name. `maybePgBossDependency`. Hm! Yes maybe, but why? Why is it maybe?
If it was `isPgBossNeeded` and returned True and False and then used in that fashion in place where deps are defined, that would be pretty direct.
Or maybe you could do `depsRequiredByJobs :: AppSpec -> [AS.Dependency.Dependency]`! What about that? That sounds great to me!

Reply:
Yup it can be hard to say sometimes, I think it really depends on the context. Here, since I started thinking more in the direction of just dependencies and not specific dependencies, it made sense to go for plural. Yeah, one way to put it is that `Maybe` is for singular, while `[]` is for plural. `Maybe` is just a special case of a list that can have 1 or 0 members.

Reply:
Yeah, transitioning from PgBoss' one dependency, to all possible of jobs, does make sense to go Maybe to []. I was originally thinking the former, but going the latter does make the second necessary. 👍🏻